discovery/discovery.go

```diff
@@ -93,12 +110,15 @@ func (o *orchestratorPool) GetOrchestrators(numOrchestrators int) ([]*net.Orches
 
 	select {
 	case <-ctx.Done():
-		glog.Info("Done fetching orch info for orchestrators, context timeout: ", orchInfos)
+		glog.Error("Done fetching orch info for orchestrators, context timeout: ", orchInfos)
 		cancel()
-		return orchInfos, nil
+		return orchInfos[:numOrchestrators], nil
```

Comment:
Do we need to index into `orchInfos` here? Seems like it would suffice to just return `orchInfos` since `getOrchInfo()` will send a signal into `orchChan` once `numOrchestrators` responses have been received (and `orchInfos` would just be empty if we timed out)

Reply:
The reason we're indexing into orchInfos here, is because often, even if `numOrchestrators` is set to one, multiple orchs are returned. This is what's happening:

> append orch1
> orchChan <- {}{}
> append orch2
> print orch list // now contains 2 orchs even though we only needed 1 

That said, since we do have a `numOrchestrators` expected, wanted to keep the returned # consistent.

Reply:
Note that it's not really an issue to return **more** orchestrators than what we asked for. However, the reason that we're locking + copying the O list is because the list may still be modified after the fact. With concurrent writes+reads, golang will crash.

That could still happen here even as the context time-out fires, so this clause should also have the same locking+copying treatment, using the original mutex.